libs/utils/WordPressUtils/src/main/java/org/wordpress/android/util/ImageUtils.java

```diff
@@ -534,44 +486,153 @@ public static String createResizedImageWithMaxWidth(Context context, String path
-            return null;
+            AppLog.e(AppLog.T.UTILS, "OutOfMemoryError while creating the resized bitmap", e);
+            throw e;
         } catch (NullPointerException e) {
             // See: https://github.com/wordpress-mobile/WordPress-Android/issues/1844
-            AppLog.e(AppLog.T.UTILS, "Bitmap.createBitmap has thrown a NPE internally. This should never happen: " + e);
-            return null;
+            AppLog.e(AppLog.T.UTILS, "Bitmap.createBitmap has thrown a NPE internally. This should never happen!", e);
+            throw e;
         }
 
         if (bmpRotated == null) {
             // Fix an issue where bmpRotated is null even if the documentation doesn't say Bitmap.createBitmap can return null.
+            AppLog.e(AppLog.T.UTILS, "bmpRotated is null even if the documentation doesn't say Bitmap.createBitmap can return null.");
             // See: https://github.com/wordpress-mobile/WordPress-Android/issues/1848
-            return null;
+            throw new IOException("bmpRotated is null even if the documentation doesn't say Bitmap.createBitmap can return null.");
         }
 
-        bmpRotated.compress(fmt, 100, stream);
-        bmpResized.recycle();
+        boolean result = bmpRotated.compress(fmt, quality, outStream);
         bmpRotated.recycle();
+        return result;
+    }
+
+
+    /**
+     * Given the path to an image, resize the image down to within a maximum width
+     * @param path the path to the original image
+     * @param maxWidth the maximum allowed width
+     * @return the path to the resized image
+     */
+    public static String createResizedImageWithMaxWidth(Context context, String path, int maxWidth) {
+        File file = new File(path);
+        if (!file.exists()) {
+            return path;
+        }
+
+        String mimeType = MediaUtils.getMediaFileMimeType(file);
+        if (mimeType.equals("image/gif")) {
+            // Don't rescale gifs to maintain their quality
+            return path;
+        }
+
+        String fileName = MediaUtils.getMediaFileName(file, mimeType);
+        String fileExtension = MimeTypeMap.getFileExtensionFromUrl(fileName).toLowerCase();
+
+        int[] dimensions = getImageSize(Uri.fromFile(file), context);
+        int orientation = getImageOrientation(context, path);
+
+        if (dimensions[0] <= maxWidth || maxWidth <= 0) {
+            // Image width is within limits; don't resize
+            return path;
+        }
+
+        Uri imageUri = Uri.parse(path);
+        if (context == null || imageUri == null) {
+            return path;
+        }
+
+        File resizedImageFile;
+        FileOutputStream out;
+
+        try {
+            resizedImageFile = File.createTempFile("wp-image-", fileExtension);
+            out = new FileOutputStream(resizedImageFile);
+        } catch (IOException e) {
+            AppLog.e(AppLog.T.POSTS, "Failed to create the temp file on storage. Use the full picture instead.");
```

Comment:
This is a static public method in `ImageUtils.java`, which should be a general purpose class. We should generalize and maybe parameterize the TAG for AppLog in the method signature instead of hardcoding `AppLog.T.POSTS` in these calls